### DISCUSSION TITLE
How to use group handlebars helper in Search Result template?

### DISCUSSION THREAD

**Discussion by Luca3082** [microsoft-search/pnp-modern-search#2967] [Q&A] [answered]
Hi, I need help...
I'm trying to create a page with PnP Search Box, Filters & Results, and I want to group by my results. I know I can use group handlebars helper but I don't know how to use it...

I'm configuring these layout slots:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/15248198/fd030008-eefd-4b4d-a120-158b1b5fa1af)
In particular, the field "Categoria" is the one I want to group by my results

Then I'm trying to use the helper in the result template, with no success...

![image](https://github.com/microsoft-search/pnp-modern-search/assets/15248198/99c01cb7-d42b-4023-aa50-18728dd1d4bf)

in row 63 I'm trying to group by Categoria and simply put the value of the field in a h1...
the next step will by using the each helper inside the group and printing the data...
any suggestion?

thank you!

**Comment by Luca3082** [comment, marked as answer]
Ok, it seems I was passing the uncorrect name of the field in the group helper...

![image](https://github.com/microsoft-search/pnp-modern-search/assets/15248198/4d279a79-bbd0-4597-af6d-4dd2ec384da0)

it seems to be the right way to start doing group by correctly... right?

**Reply by kasperbolarsen** [reply]
Which layout template are you working from? The Details List has grouping as an out of the box option, perhaps you can use that as a starting point ?

**Reply by Luca3082** [reply]
I know, but I need the Custom...
